### ISSUE TITLE
Fine-tuning with frozen stages/branches on HRNet

### ISSUE THREAD

**Issue by mgcyung** [open-mmlab/mmpose#2097]
During fine-tuning I don't want to change weights in all the
stages/branches that are in the hrnet model. How do I freeze a branch in
a stage during training for fine-tuning?

I was thinking about only freezing the first branch of each stage. How
do I accomplish this?

Any help or suggestion is greatly appreciated.

**Comment by Ben-Louis** [comment]
You could refer to https://mmengine.readthedocs.io/en/latest/tutorials/optim_wrapper.html#set-different-hyperparamters-for-different-model-modules for how to set distinct optimizer arguments for individual layers. To freeze the desired layers, consider setting their learning rate and weight decay to 0.

**Comment by mgcyung** [comment]
Thanks a lot. It really helps.
